File: sdk/js/api/http/ZLGApi.js
import ZLGAccountFlow from './httpModules/ZLGAccountFlow'
import ZLGAssistant from './httpModules/ZLGAssistant'
import ZLGCloudFile from './httpModules/ZLGCloudFile'
import ZLGCustomBigData from './httpModules/ZLGCustomBigData'
import ZLGCustomData from './httpModules/ZLGCustomData'
import ZLGDevice from './httpModules/ZLGDevice'
import ZLGDeviceCommand from './httpModules/ZLGDeviceCommand'
import ZLGDeviceData from './httpModules/ZLGDeviceData'
import ZLGDeviceEvent from './httpModules/ZLGDeviceEvent'
import ZLGDeviceGroup from './httpModules/ZLGDeviceGroup'
import ZLGDeviceType from './httpModules/ZLGDeviceType'
import ZLGFirmware from './httpModules/ZLGFirmware'
import ZLGFundAccount from './httpModules/ZLGFundAccount'
import ZLGLog from './httpModules/ZLGLog'
import ZLGMulticast from './httpModules/ZLGMulticast'
import ZLGOrganization from './httpModules/ZLGOrganization'
import ZLGPackage from './httpModules/ZLGPackage'
import ZLGProject from './httpModules/ZLGProject'
import ZLGRulesConfig from './httpModules/ZLGRulesConfig'
import ZLGRulesRecords from './httpModules/ZLGRulesRecords'
import ZLGSubDevice from './httpModules/ZLGSubDevice'
import ZLGSubuser from './httpModules/ZLGSubuser'
import ZLGSubuserAccess from './httpModules/ZLGSubuserAccess'
import ZLGUser from './httpModules/ZLGUser'
import ZLGWebApp from './httpModules/ZLGWebApp'

import ZLGHttp from './httpBase/ZLGHttp'

let ZLGApi = {
  init: function (baseURL, timeout, header) {
    ZLGHttp.getInstance(baseURL, timeout, header)
  },
  setErrorHandler: function (handler) {
    ZLGAccountFlow.setErrorHandler(handler)
    ZLGAssistant.setErrorHandler(handler)
    ZLGCloudFile.setErrorHandler(handler)
    ZLGCustomBigData.setErrorHandler(handler)
    ZLGCustomData.setErrorHandler(handler)
    ZLGDevice.setErrorHandler(handler)
    ZLGDeviceCommand.setErrorHandler(handler)
    ZLGDeviceData.setErrorHandler(handler)
    ZLGDeviceEvent.setErrorHandler(handler)
    ZLGDeviceGroup.setErrorHandler(handler)
    ZLGDeviceType.setErrorHandler(handler)
    ZLGFirmware.setErrorHandler(handler)
    ZLGFundAccount.setErrorHandler(handler)
    ZLGLog.setErrorHandler(handler)
    ZLGMulticast.setErrorHandler(handler)
    ZLGOrganization.setErrorHandler(handler)
    ZLGPackage.setErrorHandler(handler)
    ZLGProject.setErrorHandler(handler)
    ZLGRulesConfig.setErrorHandler(handler)
    ZLGRulesRecords.setErrorHandler(handler)
    ZLGSubDevice.setErrorHandler(handler)
    ZLGSubuser.setErrorHandler(handler)
    ZLGSubuserAccess.setErrorHandler(handler)
    ZLGUser.setErrorHandler(handler)
    ZLGWebApp.setErrorHandler(handler)
  },
  ZLGAccountFlow,
  ZLGAssistant,
  ZLGCloudFile,
  ZLGCustomBigData,
  ZLGCustomData,
  ZLGDevice,
  ZLGDeviceCommand,
  ZLGDeviceData,
  ZLGDeviceEvent,
  ZLGDeviceGroup,
  ZLGDeviceType,
  ZLGFirmware,
  ZLGFundAccount,
  ZLGLog,
  ZLGMulticast,
  ZLGOrganization,
  ZLGPackage,
  ZLGProject,
  ZLGRulesConfig,
  ZLGRulesRecords,
  ZLGSubDevice,
  ZLGSubuser,
  ZLGSubuserAccess,
  ZLGUser,
  ZLGWebApp
}

export default ZLGApi

File: sdk/dart/api/lib/zlg_api.dart
import 'package:flutter/material.dart';
import 'package:zlg_api/module/zlg_account_flow.dart';
import 'package:zlg_api/module/zlg_assistant.dart';
import 'package:zlg_api/module/zlg_custom_data.dart';
import 'package:zlg_api/module/zlg_device.dart';
import 'package:zlg_api/module/zlg_device_command.dart';
import 'package:zlg_api/module/zlg_device_data.dart';
import 'package:zlg_api/module/zlg_device_event.dart';
import 'package:zlg_api/module/zlg_device_group.dart';
import 'package:zlg_api/module/zlg_device_type.dart';
import 'package:zlg_api/module/zlg_project.dart';
import 'package:zlg_api/module/zlg_rule.dart';
import 'package:zlg_api/module/zlg_rule_record.dart';

import 'base/zlg_base_module.dart';
import 'module/zlg_user.dart';

export 'package:dio/dio.dart';

export 'module/zlg_user.dart';
export 'module/zlg_device.dart';
export 'module/zlg_device_data.dart';
export 'module/zlg_device_event.dart';
export 'module/zlg_device_type.dart';
export 'module/zlg_device_group.dart';
export 'module/zlg_project.dart';
export 'module/zlg_rule.dart';
export 'module/zlg_rule_record.dart';
export 'module/zlg_account_flow.dart';
export 'module/zlg_assistant.dart';
export 'module/zlg_device_command.dart';
export 'module/zlg_custom_data.dart';

export 'framework/zlg_http.dart';
export 'framework/zlg_socket.dart';

class ZLGApi extends ZLGBaseModule {
  static ZLGApi _instance;

  factory ZLGApi() {
    if (_instance == null) {
      _instance = ZLGApi._internal();
    }
    return _instance;
  }

  ZLGApi._internal();

  @override
  void setErrorHandler(ValueChanged<int> handler) {
    ZLGUser().setErrorHandler(handler);
    ZLGDevice().setErrorHandler(handler);
    ZLGDeviceData().setErrorHandler(handler);
    ZLGDeviceEvent().setErrorHandler(handler);
    ZLGDeviceType().setErrorHandler(handler);
    ZLGDeviceGroup().setErrorHandler(handler);
    ZLGProject().setErrorHandler(handler);
    ZLGRule().setErrorHandler(handler);
    ZLGRuleRecord().setErrorHandler(handler);
    ZLGAccountFlow().setErrorHandler(handler);
    ZLGAssistant().setErrorHandler(handler);
    ZLGDeviceCommand().setErrorHandler(handler);
    ZLGCustomData().setErrorHandler(handler);
  }
}
